test/transmuxer.test.js

```diff
@@ -3408,6 +3408,416 @@ QUnit.test('adjusts caption and ID3 times when configured to adjust timestamps',
   QUnit.equal(captions[0].endTime, (90004 - 90002) / 90e3, 'caption end time are based on original timeline');
 });
 
+QUnit.test('Audio frames trimmed before video, keepOriginalTimestamps = false', function() {
+  var
+    segments = [],
+    segmentLengthOnDone,
+    earliestDts = 15000,
+    transmuxer = new Transmuxer({
+      keepOriginalTimestamps: false
+    });
+
+  transmuxer.on('data', function(segment) {
+    segments.push(segment);
+  });
+  transmuxer.on('done', function(segment) {
+    if (!segmentLengthOnDone) {
+      segmentLengthOnDone = segments.length;
+    }
+  });
+
+  transmuxer.push(packetize(PAT));
+  transmuxer.push(packetize(generatePMT({
+    hasVideo: true,
+    hasAudio: true
+  })));
+
+  transmuxer.push(packetize(audioPes([
+    0x19, 0x47
+  ], true, earliestDts - 1)));
+  transmuxer.push(packetize(videoPes([
+      0x09, 0x01 // access_unit_delimiter_rbsp
+  ], true, earliestDts)));
+  transmuxer.push(packetize(videoPes([
+      0x08, 0x01 // pic_parameter_set_rbsp
+  ], true, earliestDts)));
+  transmuxer.push(packetize(videoPes([
+    0x07, // seq_parameter_set_rbsp
+    0x27, 0x42, 0xe0, 0x0b,
+    0xa9, 0x18, 0x60, 0x9d,
+    0x80, 0x53, 0x06, 0x01,
+    0x06, 0xb6, 0xc2, 0xb5,
+    0xef, 0x7c, 0x04
+  ], false, earliestDts)));
+  transmuxer.push(packetize(videoPes([
+      0x05, 0x01 // slice_layer_without_partitioning_rbsp_idr
+  ], true, earliestDts)));
+  transmuxer.flush();
+
+  QUnit.equal(segments.length, 1, 'generated a combined segment');
+  QUnit.equal(segments[0].data.length, 295, 'trimmed audio frame');
```

Comment:
What is this being compared to? Might just need a comment somewhere.

Reply:
I will add a comment